### PR TITLE
Add recaptcha (PR #132)

### LINKED ISSUE #132: Captcha ограничаване на приемането на протоколи/сигнали/снимки
https://github.com/chvarkov/google-recaptcha

https://developers.google.com/recaptcha/docs/v3

### PR DESCRIPTION
**Pull request by ogonommo**
## Какво променя този PR?

Closes #132

Dependency of ti-broish/results#47 

## Детайли

Axios and @nestjs/axios have been downgraded one version below, as @nestlab/google-recaptcha hasn't been updated to work with the latest version of the nest module. We can bump the versions again once that is done.

## Списък:

- [x] Protocol submission requires ReCAPTCHA
- [x] Violation submission requires ReCAPTCHA
- [x] Picture submission requires ReCAPTCHA

## Бележки за deployment

The Firebase app should be registered for ReCAPTCHA following [these steps](https://firebase.google.com/docs/app-check/web/recaptcha-provider).

The GOOGLE_RECAPTCHA_SECRET_KEY variable should be exposed in the running container on staging and deployment.

### CODE REVIEW

File: src/app.module.ts [hunk 1]
```diff
@@ -21,9 +21,20 @@ import { FirebaseAdminCoreModule } from './firebase/firebase-admin.module'
...
 import { CommandModule } from 'nestjs-command'
+import { GoogleRecaptchaModule } from '@nestlab/google-recaptcha'
 
 @Module({
   imports: [
+    GoogleRecaptchaModule.forRootAsync({
+      imports: [ConfigModule],
+      inject: [ConfigService],
+      useFactory: (configService: ConfigService) => ({
+        score: 0.8,
```
Comment (hkdobrev) [inline src/app.module.ts:32]: Let's move this to an env var with a default of 0.75

File: src/app.module.ts [hunk 2]
```diff
@@ -21,9 +21,20 @@ import { FirebaseAdminCoreModule } from './firebase/firebase-admin.module'
...
+        secretKey: configService.get('GOOGLE_RECAPTCHA_SECRET_KEY'),
+        response: (req) => req.headers['x-recaptcha-token'] ?? '',
+        skipIf: process.env.NODE_ENV !== 'production',
```
Comment (hkdobrev) [inline src/app.module.ts:35]: Let's move this to an explicit boolean env var `GOOGLE_RECAPTCHA_ENABLED` with a default of `false`.

File: package.json [hunk 3]
```diff
@@ -48,8 +48,9 @@
     "@nestjs/schedule": "^2.1.0",
     "@nestjs/swagger": "^6.2.1",
     "@nestjs/typeorm": "^9.0.1",
+    "@nestlab/google-recaptcha": "^3.2.0",
     "@sindresorhus/slugify": "^1.1.2",
-    "axios": "^1.3.2",
+    "axios": "^1.2.1",
```
Comment (hkdobrev) [inline package.json:53]: Why the downgrade? https://github.com/chvarkov/google-recaptcha/blob/master/package.json#L38 supports anything between v1 and v2 of axios.

**Review by hkdobrev — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (hkdobrev) [reply package.json:53]: Oh, saw your comment on that in the description.

Comment (hkdobrev) [reply package.json:53]: https://github.com/chvarkov/google-recaptcha/releases/tag/v3.2.0 should now support latest Nest 9. I don't see the discrepancy with the versions.

File: src/app.module.ts [hunk 4]
```diff
@@ -21,9 +21,20 @@ import { FirebaseAdminCoreModule } from './firebase/firebase-admin.module'
...
 
 @Module({
   imports: [
+    GoogleRecaptchaModule.forRootAsync({
+      imports: [ConfigModule],
+      inject: [ConfigService],
+      useFactory: (configService: ConfigService) => ({
+        score: 0.8,
+        secretKey: configService.get('GOOGLE_RECAPTCHA_SECRET_KEY'),
+        response: (req) => req.headers['x-recaptcha-token'] ?? '',
```
Comment (hkdobrev) [inline src/app.module.ts:34]:
```suggestion
        response: (req: Request) =>
          (req.headers['x-recaptcha-token'] as string) ?? '',
```

**Review by hkdobrev — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (ogonommo) [reply package.json:53]: It's not the Nest version itself, it's the peer dependencies that won't work. The recaptcha module supports versions of axios from v1 to v2, v1 being inlusive and v2 being exclusive, i.e. it doesn't support v2.

Comment (hkdobrev) [reply package.json:53]: oh, silly me, just noticed the change on the other line for `@nestjs/axios`. Got it now.

**Review by hkdobrev — APPROVED** [review]

**Pull request merged**